### ISSUE TITLE
Pub site search doesn't distinguish between dev dependencies and real dependencies

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/pub-dartlang-dart#1679]
URL: https://pub.dartlang.org/packages?q=dependency%3Afront_end

In the link above `kernel` should not appear - it only has a dev_dependency on the front_end which is not a real dependency. Put another way, users with a dependency on the `kernel` package would not get a dependency on the `front_end` package, only developers of the `kernel` package would.

I think it would be fine to expose querying by dev dependency, but it shouldn't be treated the same as a regular dependency as it is an entirely different thing.

**Comment by isoos** [comment]
I think this request is in conflict with the need to list packages that are using e.g. a code generator package. If you visit the package page, the Analysis tab does list the dependencies by their type, and visitors can check if it is a director, transient or dev dependency.

Putting this in the backlog, because it is likely that we'd need another notation for either the direct-only, or the direct+dev dependencies.

**Comment by sigurdm** [comment]
I agree we should not consider dev-dependencies here. We could have a separate query for that, but that is a lot less relevant for package consumers, as they are not affected by dev-dependencies.
